backend/app/controllers/spree/admin/promotions_controller.rb

```diff
@@ -28,7 +26,7 @@ def create
 
       def load_bulk_code_information
         @promotion_builder = Spree::PromotionBuilder.new(
-          base_code: @promotion.codes.first.try!(:value),
+          base_code: @promotion.base_code || @promotion.codes.first.try!(:value),
```

Comment:
to be more clear: Prior to this PR, if you look at an existing Promotion, the _first_ promotion code (if any) is shown on screen labelled (labelled incorrectly) "base code". 

That's just a mess, but the actual base code was not stored anywhere to display instead. After this change, the actual base code is stored somewhere and can be displayed properly. But for backwards compatibility and with existing promotions that do not have a stored base code, it still falls through to the old behavior. 

As I warn in the PR, this change leaves a _lot_ of internal API (class/method) API being kind of confusing. A lot of existing design is... not the choices I would have made. But I intentionally made the smallest possible change to get a properly stored base_code. 

If people would rather I change more to make the internal API more reasonable, I certainly can -- but it's very challenging to figure out how to do so in a way that is both backwards compatible and makes the code less rather than more confusing.